src/index.ts

```diff
@@ -70,22 +70,19 @@ type InitPreContextOut<T> = T extends HasInitPreContext<any, any>
   ? ReturnType<T['initPreContext']>
   : never;
 
-type SanitizeParamsContextIn<T> = T extends HasSanitizeParams<any, any>
-  ? Parameters<T['sanitizeParams']>[0]
-  : never;
-type SanitizeParamsContextOut<T> = T extends HasSanitizeParams<any, any>
-  ? ReturnType<T['sanitizeParams']>
-  : never;
-
 type AttachDataIn<T> = T extends HasAttachData<any, any>
   ? Parameters<T['attachData']>[0]
   : never;
 type AttachDataOut<T> = T extends HasAttachData<any, any>
   ? PromiseResolveOrSync<ReturnType<T['attachData']>>
   : never;
 
```

Comment:
Seems you missed some like `TRight extends HasInitPreContext<any, any>` below - and also in all the other lifecycles.
- [ ] consistency in [workaroundery]
￼

Reply:
done

Reply:
done